
# HG changeset patch
# User John H. Robinson, IV <jaqque@sbih.org>
# Date 1229206923 28800
# Node ID 9d3cac24b05fe672f821aa14d3dbed1a1cff1890
# Parent ceef975fe4e697423f61314b3fa5cf0d44d7a21c
Check for argc==0 when calling wtokensplit()

The WINGs-function wtokensplit does not set argv to NULL if no string has been
split - instead argc is set to 0.

You can only observe this issue if you compile Window Maker without any
optimization:

compile Window Maker with CFLAGS=""
run Window Maker and save a session
change session-file and replace a 'command xyz' line with 'command " "'
restart Window Maker
watch "Fatal error"-message

Author: Tobias Stoeckmann
Retrieved from: http://paldium.homeunix.org/tobias/wmaker/
Submitted by Gilbert Ashley <amigo@ibiblio.org>

--- a/src/dock.c	Sat Dec 13 14:18:06 2008 -0800
+++ b/src/dock.c	Sat Dec 13 14:22:03 2008 -0800
@@ -2977,7 +2977,7 @@
 
     wtokensplit(cmdline, &argv, &argc);
 
-    if (argv==NULL) {
+    if (!argc) {
         if (cmdline)
             wfree(cmdline);
         if (state)
--- a/src/session.c	Sat Dec 13 14:18:06 2008 -0800
+++ b/src/session.c	Sat Dec 13 14:22:03 2008 -0800
@@ -381,7 +381,7 @@
 
     wtokensplit(command, &argv, &argc);
 
-    if (argv==NULL) {
+    if (!argc) {
         return 0;
     }
 

